### PR TITLE
[Messaging] macOS keychain auth prompt fix

### PR DESCRIPTION
**Pull request by aashishpatil-g**
This is fix for Messaging to match change made to core here -> https://github.com/google/GoogleUtilities/pull/75

The fix is to avoid auth alerts (popups) on macOS when using Keychain.

Background: go/firebase-macos-keychain-popups

### CODE REVIEW

**Review by ncooke3 — APPROVED** [review]
LGTM when CI finished and is all green. Feel free to merge then!

No need to update the changelog in this PR. I'm going to do a sweep of all the affected product's changelogs and send a PR for them altogether.

Thanks!

**Comment by ncooke3** [comment]
And pls wait for @chliangGoogle's review before merging as well!

**Review by chliangGoogle — APPROVED** [review]
LGTM, make sure the issue is not reproducible after the change.

**Comment by chliangGoogle** [comment]
@aashishpatil-g Can you update the [changelog](https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseMessaging/CHANGELOG.md) too?    Use "Unreleased" at the top if you are not sure which version to go out.

**Comment by ncooke3** [comment]
> No need to update the changelog in this PR. I'm going to do a sweep of all the affected product's changelogs and send a PR for them altogether.

@chliangGoogle, I told aashishpatil-g@ to hold off on CHANGELOG update since I'm going to send a PR to update all affected CHANGELOGs after this merges. I want to coordinate the wording with morganchen12@ since we likely will include in our internal docs.

**Comment by ncooke3** [comment]
@aashishpatil-g, the failing CI is expected. The workflow in question is failing because we are testing on [all platforms](https://github.com/firebase/firebase-ios-sdk/blob/master/.github/workflows/messaging.yml#L50). 

One inevitable downside to this solution is that we cannot test macOS test cases that directly use the keychain. The fix here is to `#if` out the offending tests on macOS like what was done here: https://github.com/firebase/firebase-ios-sdk/pull/10146

File: FirebaseMessaging/Tests/UnitTests/FIRMessagingAuthKeychainTest.m
Comment (ncooke3) [inline FirebaseMessaging/Tests/UnitTests/FIRMessagingAuthKeychainTest.m:217]: Rather than `#if` out the test implementation, could you regroup the tests so they can be excluded with a single  `#if`:
```objc
// Skip keychain tests on Catalyst and macOS. Tests are skipped because they
// involve interactions with the keychain that require a provisioning profile.
// See go/firebase-macos-keychain-popups for more details.
#if !TARGET_OS_MACCATALYST && !TARGET_OS_OSX

- (void)testQueryCachedKeychainItems { ... }

- (void)testCachedKeychainOverwrite { ... }

- (void)testSetKeychainItemShouldDeleteOldEntry { ... } 

- (void)testQueryAndAddEntry { ... }


#endif  // !TARGET_OS_MACCATALYST && !TARGET_OS_OSX
```

This is for two reasons, one to reduce the number of `#ifs` we use and two so, in the event we are ever debugging test logs, the only tests that show up were actually run.

Comment (aashishpatil-g) [reply FirebaseMessaging/Tests/UnitTests/FIRMessagingAuthKeychainTest.m]: Made the change in the latest commit - afcd2c7dbc6b980127b716ac159089ef06f48612

**Review by paulb777 — APPROVED** [review]

**Review by ncooke3 — APPROVED** [review]

**Pull request merged**
